hivemind/optim/experimental/state_averager.py

```diff
@@ -376,17 +376,20 @@ def step(
         if wait_for_delayed_updates:
             for pending_update in self.pending_updates:
                 try:
+                    timeout = (averaging_opts or {}).get("averaging_timeout", self._allreduce_timeout)
                     logger.log(self.status_loglevel, "Waiting for delayed updates to finish...")
-                    output = pending_update.result()
+                    output = pending_update.result(timeout)
                 except BaseException:
-                    pass  # exception will be reported below
+                    # exception will be reported below
+                    if not pending_update.done():
+                        pending_update.cancel()
 
         # remove finished updates, log any exceptions
         finished_updates = {pending_update for pending_update in self.pending_updates if pending_update.done()}
         self.pending_updates = {pending_update for pending_update in self.pending_updates if not pending_update.done()}
         for finished_update in finished_updates:
-            if finished_update.exception():
-                logger.log(self.status_loglevel, f"Background update failed with {finished_update.exception()}")
+            if finished_update.cancelled() or finished_update.exception():
+                logger.log(self.status_loglevel, f"Background update failed: {finished_update}")
```

Comment:
Will the traceback for that be posted somewhere?

Reply:
Depending on what happened, it will be printed by state_averager or ignored (e.g. if we cancel, it will just be ignored)

Reply:
Okay.